src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs

```diff
@@ -146,6 +146,32 @@ internal static MsalTokenResponse CreateFromiOSBrokerResponse(Dictionary<string,
             return response;
         }
 
+        internal static MsalTokenResponse CreateFromAppProviderResponse(ExternalTokenResult tokenResponse)
+        {
+            var response = new MsalTokenResponse
+            {
+                AccessToken = tokenResponse.RawAccessToken,
+                RefreshToken = null,
+                IdToken = null,
+                TokenType = BrokerResponseConst.Bearer,
+                CorrelationId = tokenResponse.CorrelationId,
+                ExpiresIn = tokenResponse.ExpiresInSeconds - DateTimeHelpers.CurrDateTimeInUnixTimestamp(),
```

Comment:
I still don't understand why you need to substract `DateTimeHelpers.CurrDateTimeInUnixTimestamp()`

ExpiresInSeconds is a number like 3600
`DateTimeHelpers.CurrDateTimeInUnixTimestamp() is the numebr of seconds between Jan 1, 1970 and now, 1646389878 at the time of writing. 
Result will be negative.

This indicates that you are missing a unit test ;)